Scripts/LaunchSidechainMasternode.ps1

```diff
@@ -0,0 +1,484 @@
+if ( -not ( Get-Item -Path $mainChainDataDir -ErrorAction SilentlyContinue ) )
+{
+    New-Item -ItemType Directory -Path $mainChainDataDir
+}
+
+if ( -not ( Get-Item -Path $sideChainDataDir -ErrorAction SilentlyContinue ) )
+{
+    New-Item -ItemType Directory -Path $sideChainDataDir
+}
+
+#Gather Federation Detail
+if ( -not ( Test-Path $sideChainDataDir\federationKey.dat ) ) 
+{
+    $miningDAT = Read-Host "Please Enter the full path to the federationKey.dat"
+    Copy-Item $miningDAT -Destination $sideChainDataDir -Force -ErrorAction Stop
+}
+
+#Establish Node Type
+if ( $multiSigMnemonic -ne $null )
+{
+    $NodeType = "50K"
+}
+
+#Check for Completion
+if ( $NodeType -eq "50K" )
+{
+    if ( -not ( $multiSigPassword ) ) { $varError = 1 }
+}
+    Else
+    {
+        if ( -not ( $miningPassword ) ) { $varError = 1 }
+    }
+
+if ( -not ( $mainChainDataDir )  ) { $varError = 1 }
+if ( -not ( $sideChainDataDir )  ) { $varError = 1 }
+if ( -not ( Test-Path $sideChainDataDir/federationKey.dat ) ) { $varError = 1 }
+
+if ( $varError -eq '1' )  
+{
+    Write-Host "ERROR: Some Values were not set. Please re-run this script" -ForegroundColor Red
+    Start-Sleep 30
+    Exit
+}
+
+#Clear Host
+Clear-Host
+
+#Check for an existing running node
+Write-Host "Checking for running Mainchain Node" -ForegroundColor Cyan
+if ( Test-Connection -TargetName 127.0.0.1 -TCPPort $mainChainAPIPort )
+{
+    Write-Host "WARNING: A node is already running, will perform a graceful shutdown" -ForegroundColor DarkYellow
+    Write-Host ""
+    Shutdown-MainchainNode
+}
+
+Write-Host "Checking for running Sidechain Node" -ForegroundColor Cyan
+if ( Test-Connection -TargetName 127.0.0.1 -TCPPort $sideChainAPIPort )
+{
+    Write-Host "WARNING: A node is already running, will perform a graceful shutdown" -ForegroundColor DarkYellow
+    Write-Host ""
+    Shutdown-SidechainNode
+}
+
+#Check for running dashboard
+Write-Host "Checking for the Stratis Masternode Dashboard" -ForegroundColor Cyan
+if ( Test-Connection -TargetName 127.0.0.1 -TCPPort 37000 -ErrorAction SilentlyContinue )
+{
+    Write-Host "WARNING: The Stratis Masternode Dashboard is already running, will perform a graceful shutdown" -ForegroundColor DarkYellow
+    Write-Host ""
+    Shutdown-Dashboard
+}
+
+Write-Host ""
+
+#Check Time Difference
+Check-TimeDifference
+
+if ( $NodeType -eq "50K" ) 
+{
+    #Move to CirrusPegD
+    Set-Location -Path $cloneDir/src/Stratis.CirrusPegD
+}
+    Else
+    {
+        #Move to CirrusMinerD
+        Set-Location -Path $cloneDir/src/Stratis.CirrusMinerD
+    }
+
+#Start Mainchain Node
+$API = $mainChainAPIPort
+Write-Host "Starting Mainchain Masternode" -ForegroundColor Cyan
+if ( $NodeType -eq "50K" ) 
+{
+    $StartNode = Start-Process dotnet -ArgumentList "run -mainchain -addressindex=1 -apiport=$mainChainAPIPort -counterchainapiport=$sideChainAPIPort -redeemscript=""$redeemscript"" -publickey=$multiSigPublicKey -federationips=$federationIPs" -PassThru
```

Comment:
Please specify the build configuration so that we run Release version of the code instead of default Debug one.
`run -c Release -- -mainchain`

Reply:
Done